### PR TITLE
[WIP] Don't show 999999.0 value on CGM when a line or transformer has no Imax

### PR DESCRIPTION
**Pull request by NajjariBelgacem**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java [hunk 1]
```diff
@@ -812,12 +807,12 @@ private static void setSwitchCurrentLimit(UcteLine ucteLine, Switch sw) {
             try {
                 ucteLine.setCurrentLimit(Integer.parseInt(sw.getProperty(CURRENT_LIMIT_PROPERTY_KEY)));
             } catch (NumberFormatException exception) {
-                ucteLine.setCurrentLimit((int) DEFAULT_MAX_CURRENT);
-                LOGGER.warn("Switch {}: No current limit, set value to {}", sw.getId(), DEFAULT_MAX_CURRENT);
+                ucteLine.setCurrentLimit(null);
+                LOGGER.warn("Switch {}: No current limit, set value to {}", sw.getId(), null);
```
Comment (murgeyseb) [inline ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java:811]: As the current limit is not set, you can replace the message. For example:
No current limit provided

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/util/UcteConstants.java [hunk 2]
```diff
@@ -24,6 +24,6 @@ private UcteConstants() {
     public static final String NOT_POSSIBLE_TO_IMPORT = "It's not possible to import this network";
     public static final String ORDER_CODE = "orderCode";
     public static final String POWER_PLANT_TYPE_PROPERTY_KEY = "powerPlantType";
-    public static final double DEFAULT_MAX_CURRENT = 999999.0;
+    public static final Integer DEFAULT_MAX_CURRENT = 999999;
```
Comment (murgeyseb) [inline ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/util/UcteConstants.java:27]: Is this constant still needed now ?

**Review by murgeyseb — CHANGES_REQUESTED** [review]

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java [hunk 3]
```diff
@@ -837,15 +837,22 @@ private static UctePowerPlantType energySourceToUctePowerPlantType(Generator gen
...
-            return (int) Double.min(permanentLimit1.get(), permanentLimit2.get());
-        } else if (permanentLimit1.isPresent()) {
-            return permanentLimit1.get().intValue();
-        } else if (permanentLimit2.isPresent()) {
-            return permanentLimit2.get().intValue();
+        if (permanentLimit1.isPresent()) {
+            if (permanentLimit2.isPresent()) {
+                return (int) Double.min(permanentLimit1.get(), permanentLimit2.get());
+            }
+            if (!permanentLimit2.isPresent()) {
```
Comment (murgeyseb) [inline ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java:844]: else !

File: ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java [hunk 4]
```diff
@@ -837,15 +837,22 @@ private static UctePowerPlantType energySourceToUctePowerPlantType(Generator gen
...
+                return permanentLimit1.get().intValue();
+            }
         } else {
-            return null;
+            if (permanentLimit2.isPresent()) {
+                return permanentLimit2.get().intValue();
+            }
+            if (!permanentLimit2.isPresent()) {
```
Comment (murgeyseb) [inline ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java:851]: else !

**Review by murgeyseb — CHANGES_REQUESTED** [review]

**Review by mathbagu — CHANGES_REQUESTED** [review]
Looking to the UCTE-DEF specification, the current limit of a line or a two windings transformer is mandatory. Could you quote the specification you use to change this behavior.

I'm also suprized that there are no unit-tests in that PR. Could you please provide at least a test case to show that everything works properly if there is no current limits for a line and a 2WT.

**Comment by murgeyseb** [comment]
For the content of the PR, Document "Quality of datasets and calculation" from ENTSO-E is not accorded to what is in the UCTE format description.

It's true that in UCTE format document nothing says that Imax is optional. However, in "Quality of datasets and calculation", it is stated those requirements:
- DATA-LINE-IMAX-02 : Line max current limit / WARNING / Equivalent line current limit must be
positive or blank
- DATA-LINE-IMAX-03 : Line max current limit / WARNING / Busbar coupler current limit must
be positive or blank
- DATA-TRF-IMAX-02: Transformer current limit (non-regulated winding) / WARNING / Equivalent transformer current limit
must be positive or blank

That may explains why in real data we get elements without imax value. And we are not supposed to modify users data.

**Comment by murgeyseb** [comment]
I just realised that this PR is an old one (not closed). The actual merged one is here (#1465)

**Pull request closed without merging**